docs_src/V2TopLevelMigration.md

```diff
@@ -0,0 +1,165 @@
+- [Custom Applications Service](#custom-applications-service)
+- [Security](#security)
+
+## Custom Compose File
+
+The compose files for V2 have many changes from their V1 counter parts. If you have customized a V1 compose file to add additional services or tweak overrides,  it is highly recommended that you start with the appropriate V2 compose file and re-add your customizations. It is very likely that the sections for your additional services will need to be migrated to have the proper environment overrides. Best approach is to use one of the V2 service sections that closest matches your service  as a template.
+
+The latest V2 compose files can be found here: [https://github.com/edgexfoundry/edgex-compose/tree/ireland](https://github.com/edgexfoundry/edgex-compose/tree/ireland)
+
+### Compose Builder
+
+If the add on service(s) in your custom compose file are EdgeX released device or app services, it is highly recommended that you use the Compose Builder to generate your custom compose file. 
+
+The latest V2 Compose Builder can be found here: [https://github.com/edgexfoundry/edgex-compose/tree/ireland/compose-builder#readme](https://github.com/edgexfoundry/edgex-compose/tree/ireland/compose-builder#readme)
+
+## Database
+
+There currently is no migration path for the data stored in the database. The V2 data collections are stored separately from the V1 data collections in the Redis database. Redis is now the only supported database, i.e. support for Mongo has been removed.
+
+!!! note
+    Since the V1 data and V2 data are stored separately, one could create a migration tool and upstream it to the EdgeX community.
+
+!!! warning
+    If the database is not cleared before starting the V2 services, the old V1 data will still reside in the database taking up useful memory. It is recommended that you first wipe the database clean before starting EdgeX 2.0 Services. That is unless you create a DB migration tool. See [Clearing Redis Database](#clearing-redis-database) section below for details on how to clear the Redis database.
+
+The following sections describe what you need to be aware for the different services that create data in the database.
+
+### Core Data
+
+The Event/Reading data stored by Core Data is considered transient and of little value once it has become old. The V2 versions of these data collections will be empty until new Events/Readings are received from V2 Device Services. 
+
+The V1 ValueDescriptors have been removed in V2.
+
+### Core Metadata
+
+Most of the data stored by Core Metadata will be recreated when the V2 versions of the Device Services start-up. The statically declared devices will automatically be created and device discovery will find and add existing devices. Any device profiles, devices, provision watchers created manually via the V1 REST API will have to be recreated using the V2 REST API. Any manually-applied `AdministrativeState` settings will also need to be re-applied.
```

Comment:
Should REST API be plural in this context?

Reply:
Yep